### PR TITLE
Adds ValidateBasic for key assignment related state in provider genesis

### PR DESCRIPTION
**Pull request by danwt**
# Description

Adds a ValidateBasic util function in /provider/types for KeyAssignment data. Uses it in ValidateGenesis. 

It's a bit verbose, but simple. Note that the vscid's are not checked! They are uints

## Linked issues

- https://github.com/cosmos/interchain-security/issues/645

## Type of change

If you've checked more than one of the first three boxes, consider splitting this PR into multiple PRs!

- [x] `Fix`: Changes and/or adds code behavior, specifically to fix a bug

### CODE REVIEW

File: x/ccv/provider/keeper/key_assignment.go [hunk 1]
```diff
@@ -567,3 +571,45 @@ func (k Keeper) DeleteKeyAssignments(ctx sdk.Context, chainID string) {
...
+	consumerAddrsToPrune []types.ConsumerAddrsToPrune,
+) error {
+	for _, e := range assignedKeys {
+		if strings.TrimSpace(e.ChainId) == "" {
+			return sdkerrors.Wrap(ccvtypes.ErrInvalidGenesis, "consumer chain id must not be blank")
+		}
+		if err := sdk.VerifyAddressFormat(e.ProviderAddr); err != nil {
+			return sdkerrors.Wrap(ccvtypes.ErrInvalidGenesis, fmt.Sprintf("invalid provider address: %s", e.ProviderAddr))
+		}
+		if e.ConsumerKey == nil {
```
Comment (smarshall-spitzbart) [inline x/ccv/provider/keeper/key_assignment.go:588]: Is there any validation you can do with the key value besides just confirming the pointer is non-nil? Maybe a helper exists in sdk or tm

File: x/ccv/provider/types/key_assignment.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	ccvtypes "github.com/cosmos/interchain-security/x/ccv/types"
+)
+
+// KeyAssignmentValidateBasic validates all the genesis state for key assignment
+// This is a utility. Key Assignment does not define any new proto types, but
+// has a lot of nested data.
+func KeyAssignmentValidateBasic(
```
Comment (smarshall-spitzbart) [inline x/ccv/provider/types/key_assignment.go:15]: Duplicate function?

**Review by smarshall-spitzbart — COMMENTED** [review]
Looks like you may have unintentionally added the validation function in two different places? The logic looks good tho, just one comment

**Review by smarshall-spitzbart — APPROVED** [review]
Great stuff overall! The duplicate function should be quick to remove so I'll approve

**Comment by danwt** [comment]
Hey thanks @smarshall-spitzbart , I was doing this last thing on Friday I probably should've kept it as draft. Will finish off now. Thanks for the fixes!

**Review by mpoke — APPROVED** [review]

File: x/ccv/provider/keeper/key_assignment.go (hunk 1 above)
Comment (danwt) [reply x/ccv/provider/keeper/key_assignment.go]: Hmm I'm not sure to be honest. It's basic validation only, a full semantic validation that all the data is consistent logically isn't the goal here so we can probably leave it like this.

**Pull request merged**